enterprise_access/apps/content_assignments/api.py

```diff
@@ -355,3 +356,25 @@ def cancel_assignments(assignments: Iterable[LearnerContentAssignment]) -> dict:
         'cancelled': list(set(cancelled_assignments) | already_cancelled_assignments),
         'non_cancelable': list(non_cancelable_assignments),
     }
+
+
+def get_content_metadata_for_assignments(enterprise_catalog_uuid, assignments):
+    """
+    Fetches (from cache or enterprise-catalog API call) content metadata
+    in bulk for the `content_keys` of the given assignments, provided
+    such metadata is related to the given `enterprise_catalog_uuid`.
+
+    Returns:
+        A dict mapping every content key of the provided assignments
+        to a content metadata dictionary, or null if no such dictionary
+        could be found for a given key.
+    """
+    content_keys = sorted({assignment.content_key for assignment in assignments})
+    content_metadata_list = get_and_cache_catalog_content_metadata(enterprise_catalog_uuid, content_keys)
+    metadata_by_key = {
+        record['key']: record for record in content_metadata_list
+    }
+    return {
+        assignment.content_key: metadata_by_key.get(assignment.content_key)
```

Comment:
[curious, non-blocking] Could there ever be a chance where the assignment's content key is no longer part of the catalog when this gets evaluated? How does the `get_content_metadata` API handle content keys that are not part of the catalog?

Reply:
you'd end up with the assignment's content_key mapping to null here.  The situation you're describing is technically possible (content falls out of catalog after allocation), but I'm not sure exactly what should happen in that case.  Ideally we'd have a hook in place to error out any unaccepted assignments.
The `get_content_metadata` endpoint won't include a record for requested keys that are not in the catalog, but the response is still a 200 as long as the catalog actually exists (even when *no* requested keys are found within it).